rclpy/test/talker.py

```diff
@@ -20,34 +22,78 @@
 sys.path.insert(0, os.getcwd())
 
 
-def talker():
+def fill_msg(msg, message_name, i):
+    if message_name == 'String':
+        msg.data = 'Hello World: {}'.format(i)
+    elif message_name == 'TransformStamped':
+        msg.transform.translation.x = float(i)
+        msg.transform.translation.y = float(i + 1)
+        msg.transform.translation.z = float(i + 2)
+        msg.transform.rotation.x = float(i + 3)
+        msg.transform.rotation.y = float(i + 4)
+        msg.transform.rotation.z = float(i + 5)
+        msg.transform.rotation.w = float(i + 6)
+    elif message_name == 'Imu':
+        msg.angular_velocity_covariance = [float(x) for x in range(i, i + 9)]
+    elif message_name == 'LaserScan':
+        msg.intensities = [float(x) for x in range(i + 10)]
+    elif message_name == 'PointCloud2':
+        from sensor_msgs.msg import PointField
+        msg.fields = []
+        for x in range(i + 2):
+            tmpfield = PointField()
+            tmpfield.name = 'toto' + str(x)
+            msg.fields.append(tmpfield)
+    else:
+        raise NotImplementedError('no test coverage for {}'.format(message_name))
+    return msg
+
+
+def talker(message_pkg, message_name, rmw_implementation, number_of_cycles):
     import rclpy
     from rclpy.qos import qos_profile_default
 
     rclpy.init([])
 
     # TODO(wjwwood) move this import back to the module level when
     # it is possible to import the messages before rclpy.init().
-    from std_msgs.msg import String
+    module = importlib.import_module(message_pkg + '.msg')
+    msg_mod = getattr(module, message_name)
+    assert msg_mod.__class__._TYPE_SUPPORT is not None
 
     node = rclpy.create_node('talker')
 
-    chatter_pub = node.create_publisher(String, 'chatter', qos_profile_default)
+    chatter_pub = node.create_publisher(msg_mod, 'chatter', qos_profile_default)
 
-    msg = String()
+    msg = msg_mod()
 
-    i = 1
+    msg_count = 1
     print('talker: beginning loop')
-    while True:
-        msg.data = 'Hello World: {0}'.format(i)
-        i += 1
-        print('talker sending: ' + msg.data)
+    while rclpy.ok() and msg_count < number_of_cycles:
+        msg = fill_msg(msg, message_name, msg_count)
+        msg_count += 1
         chatter_pub.publish(msg)
+        print('talker sending: %r' % msg)
         time.sleep(1)
+    rclpy.shutdown()
 
 if __name__ == '__main__':
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('-p', '--message_pkg', default='std_msgs',
+                        help='name of the message package')
+    parser.add_argument('-m', '--message_name', default='String',
+                        help='name of the ROS message')
+    parser.add_argument('-r', '--rmw_implementation', default='rmw_opensplice_cpp',
```

Comment:
If the system has only one rmw implementation but not OpenSplice this default is not very helpful. Should the default be selected based on the ament index instead?